pkgs/applications/editors/jupyter/default.nix

```diff
@@ -0,0 +1,21 @@
+{ stdenv
+, lib
+, pkgs
+, python3
+, python3Packages 
+, kernelDefinitions ? null
+}:
+
+with lib;
+
+let
+
+  jupyterPath = pkgs.jupyterKernels.override {kernelDefinitions = kernelDefinitions; };
+
+in
+
+python3Packages.buildPythonApplication rec {
+  inherit (python3Packages.notebook) pname version name src LC_ALL buildInputs propagatedBuildInputs checkPhase doCheck meta;
+
+  makeWrapperArgs = ["--set JUPYTER_PATH ${jupyterPath}"];
```

Comment:
```
let
 ...
in with python3.pkgs; toPythonModule (notebook.override {
  makeWrapperArgs = ["--set JUPYTER_PATH ${jupyterPath}"];
});
```

`toPythonModule (buildPythonPackage {...})` is basically the same as `buildPythonApplication`. In this case I would go for the former, because the current approach requires keeping up with required attributes.